### PR TITLE
Topics in search

### PR DESCRIPTION
**Pull request by watermelonT**
Add documentation for returning topics in search API.

### CODE REVIEW

File: api-reference/beta/resources/search-topic.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+|Method|Return type|Description|
+|:---|:---|:---|
+|[Get topic](../api/search-topic-get.md)|[microsoft.graph.search.topic](../resources/search-topic.md)|Read the properties and relationships of a [topic](../resources/search-topic.md) object.|
+
+## Properties
+
+|Property|Type|Description|
+|:---|:---|:---|
+| id (inherited) | Edm.String | The unique identifier of the topic. |
+| primaryName | microsoft.graph.topicName | The object representing the most relevant name and its metadata. |
```
Comment (ajacks-msft) [inline api-reference/beta/resources/search-topic.md:29]: "most relevant" is fairly internal data science wording. Shall we say "most accurate"? Also on the next line.

File: api-reference/beta/resources/search-topic.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+|:---|:---|:---|
+| id (inherited) | Edm.String | The unique identifier of the topic. |
+| primaryName | microsoft.graph.topicName | The object representing the most relevant name and its metadata. | 
+| alternateNames | Collection(microsoft.graph.topicName) | A list of objects representing all other relevant names and their metadata. | 
+| definition | microsoft.graph.topicDefinition | The object representing a short description of the topic. | 
+| lifecycle | microsoft.graph.topicLifecycle | The lifecycle of the topic, whether a topic is suggested, confirmed, or published. | 
+| viewpoint | microsoft.graph.topicViewpoint | Represents user viewpoints data on the topic. The data includes the user's feedback on the topic item. | 
+| webUrl | Edm.String | The URL of the topic directing viewer to the topic page in SharePoint topic center. | 
+| currentLanguagueTag | Edm.String | The language tag of the current topic view |
+| languageInfos | Collection(microsoft.graph.topicLanguageInfo) | Other available language infos of the current topic, with language name and the URL to visit topic page using given language. |
```
Comment (ajacks-msft) [inline api-reference/beta/resources/search-topic.md:36]: I don't understand what a language info is. Is there a more descriptive way to document this?

File: api-reference/beta/resources/search-topic.md [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+| viewpoint | microsoft.graph.topicViewpoint | Represents user viewpoints data on the topic. The data includes the user's feedback on the topic item. | 
+| webUrl | Edm.String | The URL of the topic directing viewer to the topic page in SharePoint topic center. | 
+| currentLanguagueTag | Edm.String | The language tag of the current topic view |
+| languageInfos | Collection(microsoft.graph.topicLanguageInfo) | Other available language infos of the current topic, with language name and the URL to visit topic page using given language. | 
+| highlyRelatedFiles | Collection(microsoft.graph.topicFile) | The list hosting at most 10 files mined or curated that are highly relevant for this topic. |
+| highlyRelatedSites | Collection(microsoft.graph.topicSite) | The list hosting at most 10 sites mined or curated that are highly relevant for this topic. |
+| highlyRelatedUsers | Collection(microsoft.graph.topicUser) | The list hosting at most 10 users mined or curated that are highly relevant for this topic. |
+| lastModifiedBy | microsoft.graph.identity | The last modifier of the topic, if the topic has been modified |
```
Comment (ajacks-msft) [inline api-reference/beta/resources/search-topic.md:40]: "The last user to edit this topic" would be more clear. Or you use "curator" later in this doc.

File: api-reference/beta/resources/search-topic.md [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+
+### File
+File inherits from microsoft.graph.topicResourceBase, it has an webUrl field hosting URL property to uniquely identify a file, it has an displayName hosting the display name of the file, as well as a list of properties in microsoft.graph.topicResourceBase describing how the file is associated with the topic.
+
+#### Properties
+| Property       | Type           | Description                                 | 
+| -------------- | -------------- | ------------------------------------------- | 
+| webUrl | Edm.String | The URL of file item. | 
+| displayName | Edm.String | The display name of the file item. |
+| languageTags | Collection(Edm.String) | The list hosting a set of language tags mined for this file. |
```
Comment (ajacks-msft) [inline api-reference/beta/resources/search-topic.md:156]: What format are language tags in? Is this like "en-us"?

File: api-reference/beta/resources/search-topic.md [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+| lifecycle | microsoft.graph.topicLifecycle | The lifecycle of the file, whether a file is suggested, confirmed, or published. |
+| viewpoint | microsoft.graph.topicViewpoint | Represents user viewpoints data on the file. The data includes the user's feedback on the file item. |
+| lastModifiedBy | microsoft.graph.identity | The last curator of the file, if the file has been curated |
```
Comment (ajacks-msft) [inline api-reference/beta/resources/search-topic.md:159]: Would it help to say "manually curated"?

**Review by ajacks-msft — APPROVED** [review]
Very helpful doc! My comments are all minor.

File: api-reference/beta/api/topicsearch-post.md [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+## Optional query parameters
+This method supports the [OData Query Parameters](/graph/query-parameters) to help customize the response.
+
+## Request headers
+| Name      |Description|
+|:----------|:----------|
+| Authorization  | Bearer {token}. Required. |
+
+## Request body
+Do not supply a request body for this method.
```
Comment (njerigrevious) [inline api-reference/beta/api/topicsearch-post.md:43]: Why does it say "do not supply a request body for this method"? Shouldn't  users be able to provide a querystring that will be used to find relevant topics that will be displayed in the response? Please provide more info on how this works and the intended use of the request body for this feature. Thanks!

**Review by njerigrevious — CHANGES_REQUESTED** [review]
Looks good - would like some additional clarification on the request body mechanism per my comment. Thanks!

**Pull request closed without merging**